### PR TITLE
Create adapter package for data and child change listener and prepare zkclient

### PR DESCRIPTION
**Pull request by qqu0127**
### Issues

- [ ] My PR addresses the following Helix issues and references them in the PR description:

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Prepare zkclient and implement new adapter for child change listener.
**The real implementation of child change listener in meta-client is left for future PR.**

Similar as https://github.com/apache/helix/pull/2333, this PR adds a new default method in `IZkChildListener` to make it applicable to be used in meta-client. Implement a adapter class that bridges the old and new listener interface.
Move previous inner converter class to separate module. 

### Tests

- [X] The following tests are written for this issue:

Change in zookeeper-api and meta-client are both backward compatible, covered by existing tests.

- The following is the result of the "mvn test" command on the appropriate module:

[INFO] Tests run: 67, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1,044.795 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 67, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.6:report (generate-code-coverage-report) @ zookeeper-api ---
[INFO] Loading execution data file /Users/qqu/workspace/qqu-helix/zookeeper-api/target/jacoco.exec
[INFO] Analyzed bundle 'Apache Helix :: ZooKeeper API' with 116 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  17:40 min
[INFO] Finished at: 2023-01-20T11:36:57-05:00
[INFO] ------------------------------------------------------------------------

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Review by junkaixue — APPROVED** [review]
Overall, looks good to me.

File: meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/adapter/ChildListenerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  private final ChildChangeListener _listener;
+
+  public ChildListenerAdapter(ChildChangeListener listener) {
+    _listener = listener;
+  }
+
+  @Override
+  public void handleChildChange(String parentPath, List<String> currentChilds) throws Exception {
+    throw new UnsupportedOperationException("handleChildChange(String parentPath, List<String> currentChilds) "
+        + "is not supported");
```
[the remark this answers is not in the document]
Comment (qqu0127) [reply meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/adapter/ChildListenerAdapter.java:41]: These are not the same thing. 
This interface method is kind of deprecated in this adapter, it should never be invoked. See the change in ZkClient

File: zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/IZkChildListener.java [hunk 2]
```diff
@@ -42,4 +44,9 @@ public interface IZkChildListener {
      * @throws Exception
      */
     public void handleChildChange(String parentPath, List<String> currentChilds) throws Exception;
+
+    default void handleChildChange(String parentPath, List<String> currentChilds, Watcher.Event.EventType eventType)
```
[the remark this answers is not in the document]
Comment (qqu0127) [reply zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/IZkChildListener.java:56]: Thanks, updated.

**Comment by qqu0127** [comment]
This PR is ready to merge, approved by @junkaixue 
Commit message: 
Prepare zkclient and implement new adapter for child change listener.

**Pull request merged**